cmd/protoc-gen-openapi/generator/openapi-v3.go

```diff
@@ -103,6 +104,12 @@ func (g *OpenAPIv3Generator) buildDocumentV3() *v3.Document {
 
 	for _, file := range g.plugin.Files {
 		if file.Generate {
+			// Merge any `Document` annotations with the current
+			extDocument := proto.GetExtension(file.Desc.Options(), v3.E_Document)
+			if extDocument != nil {
+				proto.Merge(d, extDocument.(*v3.Document))
```

Comment:
Nice, this seems really clever and elegant. I wonder if there are any other places (besides methods and files) where it would make sense to attach a partial document - maybe messages or fields?

Reply:
Yeah, I can add the same for Messages=>Schemas and Fields=>Parameters. But didn't want to do too much in case you reject the whole idea. I can do this in a separate PR.

Reply:
I think that would be great, thanks!